yarpl/include/yarpl/flowable/FlowableObserveOnOperator.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+namespace yarpl {
+namespace flowable {
+namespace detail {
+
+template <typename T>
+class ObserveOnOperatorSubscriber;
+
+template <typename T>
+class ObserveOnOperatorSubscription : public yarpl::flowable::Subscription {
+ public:
+  ObserveOnOperatorSubscription(
+      Reference<ObserveOnOperatorSubscriber<T>> subscriber,
+      Reference<Subscription> subscription)
+      : subscriber_(std::move(subscriber)),
+        subscription_(std::move(subscription)) {}
+
+  // all requesting methods are called from 'executor_' in the
+  // associated subscriber
+  void cancel() override {
+    if (auto subscriber = std::move(subscriber_)) {
+      subscriber_->isCanceled_ = true;
```

Comment:
You mean `subscriber`?  `subscriber_` has been moved away.

Should we keep the `subscriber` reference till the end of the function, just in case `subscription_->cancel()` blows away the last reference to `this`?

Reply:
added a test to verify that this case is covered, good catch!
I'll keep a 'self' around, in case that happens. The call to subscription_->cancel() might also blow our last reference away, in a way that makes accessing subscriber_ unsafe as well.